File: telegram-bot.py
import requests
import json

class TelegramBot:
    def __init__(self, bot_token, chat_id):
        self.bot_token = bot_token
        self.chat_id = chat_id

    def send_message(self, text):
        try:
            response = requests.post(
                f'https://api.telegram.org/bot{self.bot_token}/sendMessage',
                headers={'Content-Type': 'application/json'},
                data=json.dumps({
                    'chat_id': self.chat_id,
                    'text': text
                })
            )
            data = response.json()
            print(data)
        except Exception as e:
            print(e)

    def delete_message(self, message_id):
        try:
            response = requests.post(
                f'https://api.telegram.org/bot{self.bot_token}/deleteMessage',
                headers={'Content-Type': 'application/json'},
                data=json.dumps({
                    'chat_id': self.chat_id,
                    'message_id': message_id
                })
            )
            data = response.json()
            print(data)
        except Exception as e:
            print(e)

    def read_messages(self):
        try:
            response = requests.post(
                f'https://api.telegram.org/bot{self.bot_token}/getUpdates',
                headers={'Content-Type': 'application/json'},
                data=json.dumps({})
            )
            data = response.json()
            if data.get('ok'):
                messages = data.get('result')
                specific_chat_messages = [message for message in messages if message.get('message', {}).get('chat', {}).get('id') == self.chat_id]
                print(specific_chat_messages)
            else:
                print('Error fetching updates:', data.get('description'))
        except Exception as e:
            print(e)

    def edit_message(self, message_id, new_text):
        try:
            response = requests.post(
                f'https://api.telegram.org/bot{self.bot_token}/editMessageText',
                headers={'Content-Type': 'application/json'},
                data=json.dumps({
                    'chat_id': self.chat_id,
                    'message_id': message_id,
                    'text': new_text
                })
            )
            data = response.json()
            if data.get('ok'):
                print('Message edited successfully:', data.get('result'))
            else:
                print('Error editing message:', data.get('description'))
        except Exception as e:
            print(e)

    def reply_to_message(self, text, message_id):
        try:
            response = requests.post(
                f'https://api.telegram.org/bot{self.bot_token}/sendMessage',
                headers={'Content-Type': 'application/json'},
                data=json.dumps({
                    'chat_id': self.chat_id,
                    'text': text,
                    'reply_to_message_id': message_id
                })
            )
            data = response.json()
            if data.get('ok'):
                print('Reply sent successfully:', data.get('result'))
            else:
                print('Error sending reply:', data.get('description'))
        except Exception as e:
            print(e)

    def send_poll(self, question, options):
        try:
            response = requests.post(
                f'https://api.telegram.org/bot{self.bot_token}/sendPoll',
                headers={'Content-Type': 'application/json'},
                data=json.dumps({
                    'chat_id': self.chat_id,
                    'question': question,
                    'options': options,
                    'is_anonymous': True,  # optional: set to False if the poll should be public
                    'type': 'regular'  # 'quiz' for a quiz
                })
            )
            data = response.json()
            if data.get('ok'):
                print('Poll sent successfully:', data.get('result'))
            else:
                print('Error sending poll:', data.get('description'))
        except Exception as e:
            print(e)

    def pin_message(self, message_id):
        try:
            response = requests.post(
                f'https://api.telegram.org/bot{self.bot_token}/pinChatMessage',
                headers={'Content-Type': 'application/json'},
                data=json.dumps({
                    'chat_id': self.chat_id,
                    'message_id': message_id
                })
            )
            data = response.json()
            if data.get('ok'):
                print('Message pinned successfully')
            else:
                print('Error pinning message:', data.get('description'))
        except Exception as e:
            print(e)

# Example Usage:
bot = TelegramBot('<YOUR_BOT_TOKEN_HERE>', '<YOUR_CHAT_ID_HERE>')
bot.send_message('Hello, world!')
bot.delete_message(<MESSAGE_ID_HERE>)
bot.read_messages()
bot.edit_message(<MESSAGE_ID_HERE>, 'New Text')
bot.reply_to_message('Your reply text', <MESSAGE_ID_HERE>)
bot.send_poll('Your poll question?', ['Option_1', 'Option_2', 'Option_3', 'Option_4'])
bot.pin_message(<MESSAGE_ID_HERE>)

File: telegram-bot.js
class TelegramBot {
    constructor(botToken, chatId) {
      this.botToken = botToken;
      this.chatId = chatId;
      this.results = {};
    }
  
    async sendMessage(text) {
      try {
        const response = await fetch(`https://api.telegram.org/bot${this.botToken}/sendMessage`, {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json'
          },
          body: JSON.stringify({
            chat_id: this.chatId,
            text: text
          })
        });
        const data = await response.json();
        console.log(data);
        this.results.sendMessage = data;
      } catch (error) {
        console.error(error);
      }
    }
  
    async deleteMessage(messageId) {
      try {
        const response = await fetch(`https://api.telegram.org/bot${this.botToken}/deleteMessage`, {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json'
          },
          body: JSON.stringify({
            chat_id: this.chatId,
            message_id: messageId
          })
        });
        const data = await response.json();
        console.log(data);
        this.results.deleteMessage = data;
      } catch (error) {
        console.error(error);
      }
    }
  
    async readMessages() {
      try {
        const response = await fetch(`https://api.telegram.org/bot${this.botToken}/getUpdates`, {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json'
          }
        });
        const data = await response.json();
        if (data.ok) {
          const messages = data.result;
          const specificChatMessages = messages.filter(message => message.message.chat.id === this.chatId);
          console.log(specificChatMessages);
            this.results.readMessages = specificChatMessages;
        } else {
          console.log('Error fetching updates:', data.description);
          this.results.readMessages = data.description;
        }
      } catch (error) {
        console.error(error);
      }
    }
  
    async editMessage(messageId, newText) {
      try {
        const response = await fetch(`https://api.telegram.org/bot${this.botToken}/editMessageText`, {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json'
          },
          body: JSON.stringify({
            chat_id: this.chatId,
            message_id: messageId,
            text: newText
          })
        });
        const data = await response.json();
        if (data.ok) {
          console.log('Message edited successfully:', data.result);
          this.results.editMessage = data.result;
        } else {
          console.error('Error editing message:', data.description);
          this.results.editMessage = data.description;
        }
      } catch (error) {
        console.error(error);
      }
    }
  
    async replyToMessage(text, messageId) {
      try {
        const response = await fetch(`https://api.telegram.org/bot${this.botToken}/sendMessage`, {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json'
          },
          body: JSON.stringify({
            chat_id: this.chatId,
            text: text,
            reply_to_message_id: messageId
          })
        });
        const data = await response.json();
        if (data.ok) {
          console.log('Reply sent successfully:', data.result);
          this.results.replyToMessage = data.result;
        } else {
          console.error('Error sending reply:', data.description);
          this.results.replyToMessage = data.description;
        }
      } catch (error) {
        console.error(error);
      }
    }
  
    async sendPoll(question, options, isAnonymous = true, type = 'regular') {
      try {
        const response = await fetch(`https://api.telegram.org/bot${this.botToken}/sendPoll`, {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json'
          },
          body: JSON.stringify({
            chat_id: this.chatId,
            question: question,
            options: options,
            is_anonymous: isAnonymous,
            type: type
          })
        });
        const data = await response.json();
        if (data.ok) {
          console.log('Poll sent successfully:', data.result);
          this.results.sendPoll = data.result;
        } else {
          console.error('Error sending poll:', data.description);
          this.results.sendPoll = data.description;
        }
      } catch (error) {
        console.error(error);
      }
    }
  
    async pinMessage(messageId) {
      try {
        const response = await fetch(`https://api.telegram.org/bot${this.botToken}/pinChatMessage`, {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json'
          },
          body: JSON.stringify({
            chat_id: this.chatId,
            message_id: messageId
          })
        });
        const data = await response.json();
        if (data.ok) {
          console.log('Message pinned successfully');
          this.results.pinMessage = 'Message pinned successfully';
        } else {
          console.error('Error pinning message:', data.description);
          this.results.pinMessage = data.description;
        }
      } catch (error) {
        console.error(error);
      }
    }
  }
